### DISCUSSION TITLE
More ffmpeg encoding tasks

### DISCUSSION THREAD

**Discussion by andanko** [mediacms-io/mediacms#862] [Q&A]
I would like to use more than 8 encoding tasks, because I'm using hardware encoding with an RTX 3060 with an unrestricted driver.
I'm not an expert and I can't find this parameter.
As you can see in the picture the VGA is using only 35%.

![viber_image_2023-08-24_15-38-54-074](https://github.com/mediacms-io/mediacms/assets/59030241/ea5b8d81-80ed-437c-aa12-de08174e84fb)

**Comment by mgogoulos** [comment]
I believe it's a matter of running ffmpeg with some extra options, but haven't done it yet. Check out if this helps

https://github.com/mediacms-io/mediacms/issues/21
https://docs.nvidia.com/video-technologies/video-codec-sdk/12.0/ffmpeg-with-nvidia-gpu/index.html
https://stackoverflow.com/questions/44510765/gpu-accelerated-video-processing-with-ffmpeg

**Comment by andanko** [comment]
Thank you.

I have tried lot of settings according to this site. Now I'm using with this parameters

 ffmpeg -y -threads 1 -hwaccel nvdec -hwaccel_output_format cuda -i /home/share/001.mkv -vf scale_cuda=if(lt(iw\,ih)\,240\,427):if(lt(iw\,ih)\,427\,240):force_original_aspect_ratio=decrease:interp_algo=lanczos -c:v h264_nvenc -gpu:v 0 -cq:v 21 -c:a aac -b:a 160k -ac 2 -rc:v vbr -tune:v ll -preset:v p5 -b:v 0 -keyint_min 239 -maxrate 450k -bufsize 360k -profile:v main -level 4.2 -f mp4 -movflags +faststart /home/share/encoded.mp4[](url)

I really can't find any other solution for accelerating the encoding speed with hardware encoding, just more simultaneous thread usage.

The ffmpeg "thread 1" parameter doesn't do anything it can be 2,4 or 8 the encoding speed still approx 85-90 fps.

![viber_image_2023-08-25_18-26-35-616](https://github.com/mediacms-io/mediacms/assets/59030241/28a154be-a43a-421e-90be-92c241dc0c4b)

**Reply by airhao3** [reply]
I do have a rtx 3070 GPU, but I don‘t know how to config these code to use it. Can you please tech me how to do it ? Plz

**Reply by acleitao-projects** [reply]
> Thank you.
> 
> I have tried lot of settings according to this site. Now I'm using with this parameters
> 
> ffmpeg -y -threads 1 -hwaccel nvdec -hwaccel_output_format cuda -i /home/share/001.mkv -vf scale_cuda=if(lt(iw,ih),240,427):if(lt(iw,ih),427,240):force_original_aspect_ratio=decrease:interp_algo=lanczos -c:v h264_nvenc -gpu:v 0 -cq:v 21 -c:a aac -b:a 160k -ac 2 -rc:v vbr -tune:v ll -preset:v p5 -b:v 0 -keyint_min 239 -maxrate 450k -bufsize 360k -profile:v main -level 4.2 -f mp4 -movflags +faststart /home/share/encoded.mp4
> 
> I really can't find any other solution for accelerating the encoding speed with hardware encoding, just more simultaneous thread usage.
> 
> The ffmpeg "thread 1" parameter doesn't do anything it can be 2,4 or 8 the encoding speed still approx 85-90 fps.
> 
> ![viber_image_2023-08-25_18-26-35-616](https://private-user-images.githubusercontent.com/59030241/263334052-28a154be-a43a-421e-90be-92c241dc0c4b.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM2MjI5ODcsIm5iZiI6MTc0MzYyMjY4NywicGF0aCI6Ii81OTAzMDI0MS8yNjMzMzQwNTItMjhhMTU0YmUtYTQzYS00MjFlLTkwYmUtOTJjMjQxZGMwYzRiLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDAyVDE5MzgwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3ZWUwODRiNGI2NmZlODBiMzcyZmI0OTMwNjQ4ZWFlMGJkMmY1MGY5N2QwYTVhYjMzYzA3MjRjMzEwMTllZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3zXvu6qcR8cxsYuIMnV1SarHQqpVZgXIbCl5uKtNfKo)

Same problem here... its slower than running on CPU... did you ifind any solution for it? Funny thing is my 1650 with only 4gb just run topping 100% of gpu utilization and memory

**Comment by TomErnst1972** [comment]
Hi,
just a question:

are you using the hardware encoding within MediaCMS? And if yes, how did you manage to integrate it?

> I would like to use more than 8 encoding tasks, because I'm using hardware encoding with an RTX 3060 with an unrestricted driver. I'm not an expert and I can't find this parameter. As you can see in the picture the VGA is using only 35%.
>
